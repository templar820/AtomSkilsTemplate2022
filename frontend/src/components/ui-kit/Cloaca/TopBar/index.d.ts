import * as React from 'react';
import { EmotionStylesType } from 'lib-ui';

export interface TopBarProps {
    /** Defines left items */
    leftItems?: Array<object>;
    /** Defines left menu title */
    leftMenuOverTitle?: string;
    /** Defines occurence */
    occurrence?: string;
    /** Defines right items */
    rightItems?: Array<object>;
    /** Defines component's additional styles */
    style?: EmotionStylesType;
    /** Defines component's additional css */
    css?: EmotionStylesType;
}

declare const TopBar: React.ComponentType<TopBarProps>;

export default TopBar;
