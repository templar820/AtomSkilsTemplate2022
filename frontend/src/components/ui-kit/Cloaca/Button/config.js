export default {
    animation: {
        duration: 500
    }
};
