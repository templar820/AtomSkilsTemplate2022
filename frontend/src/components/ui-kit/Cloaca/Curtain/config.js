export default {
    position: {
        top: {
            vertical: true,
            inverse: false
        },
        right: {
            vertical: false,
            inverse: true
        },
        bottom: {
            vertical: true,
            inverse: true
        },
        left: {
            vertical: false,
            inverse: false
        }
    }
};
