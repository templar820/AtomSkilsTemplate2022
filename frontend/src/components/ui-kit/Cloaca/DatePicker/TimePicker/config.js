import { setHours, setMinutes, setSeconds } from 'date-fns';

// in px
const ITEM_HEIGHT = 50;

const methodsDict = {
    hours: setHours,
    minutes: setMinutes,
    seconds: setSeconds
};

const HOURS_COUNT = 24;
const MINUTES_LENGTH = 60;
const SECONDS_LENGTH = 60;

export { ITEM_HEIGHT, methodsDict, HOURS_COUNT, MINUTES_LENGTH, SECONDS_LENGTH };
