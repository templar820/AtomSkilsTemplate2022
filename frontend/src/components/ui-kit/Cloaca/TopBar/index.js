import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { css } from '@emotion/core';

import { positionUtils, debounce } from 'lib-root/utils';
import { Dropdown, Item as DropdownItem } from 'lib-ui/Dropdown';

import { withColors } from 'lib-ui/utils';
import { StyledLeftMenuItem, StyledRightMenuItem, StyledShowMoreMenuItem, StyledTopBar } from './units';
import { setClassName } from 'lib-root/utils/styleMixins';

/**
 * Контейнер для выпадающих меню.
 */
class TopBar extends Component {
    topBarRef = React.createRef();
    menuItemsRef = {
        left: [],
        more: [],
        right: []
    };

    state = {
        lastShowedMenuIndex: Infinity
    };

    renderLeftMenu = (items, colors) =>
        items.map((item, index) => {
            if (!item.props.disabled) {
                return (
                    <StyledLeftMenuItem
                        {...{ colors }}
                        isHidden={index > this.state.lastShowedMenuIndex}
                        key={index}
                        ref={(menuItem) => (this.menuItemsRef.left[index] = menuItem)}>
                        {item}
                    </StyledLeftMenuItem>
                );
            }
            return null;
        });

    renderShowMoreMenu = (items, colors) =>
        items.map((item, index) => (
            <StyledShowMoreMenuItem
                {...{ colors }}
                isHidden={this.state.lastShowedMenuIndex >= Object.keys(this.menuItemsRef.left).length}
                key={index}
                ref={(menuItem) => (this.menuItemsRef.more[index] = menuItem)}>
                {item}
            </StyledShowMoreMenuItem>
        ));

    renderRightMenu = (items, colors) => {
        if (items.length) {
            return items.map(
                (item, index) =>
                    !item.props.disabled && (
                        <StyledRightMenuItem
                            {...{ colors }}
                            key={index}
                            ref={(menuItem) => (this.menuItemsRef.right[index] = menuItem)}>
                            {item}
                        </StyledRightMenuItem>
                    )
            );
        }
        /* if right menu item doesn't exist, render blank div */
        return <div ref={(menuItem) => (this.menuItemsRef.right[0] = menuItem)}> </div>;
    };

    mapItems = (array, wrap) =>
        array.map((element) => (
            <DropdownItem
                key={element.text}
                occurrence={wrap ? this.props.occurrence : ''}
                leftIcon={element.leftIcon}
                rightIcon={element.rightIcon}
                childDropdown={
                    element.childDropdown && <Dropdown>{this.mapItems(element.childDropdown, false)}</Dropdown>
                }
                childDropdownPlacement={wrap ? 'bottom' : 'right-start'}
                childWrapperStyle={
                    wrap
                        ? css`
                              padding-top: 5px;
                          `
                        : undefined
                }
                value={element.value}
                onClick={element.onClick || (() => {})}
                disabled={element.disabled}>
                {element.text}
            </DropdownItem>
        ));

    getMaxLeftItems = () => {
        const { menuItemsRef, topBarRef } = this;
        const { lastShowedMenuIndex } = this.state;

        if (!(topBarRef && menuItemsRef.left.length && menuItemsRef.more.length && menuItemsRef.right.length)) {
            return;
        }

        const positionOfTopBar = positionUtils.getPositionOnPage(topBarRef.current);
        const positionOfRightMenu = positionUtils.getPositionOnPage(menuItemsRef.right[0]);
        const positionOfMoreItem = positionUtils.getPositionOnPage(menuItemsRef.more[0]);

        if (
            positionOfRightMenu.right > positionOfTopBar.right - 8 ||
            lastShowedMenuIndex < Object.keys(menuItemsRef.left).length
        ) {
            let lastIndex = -1;
            let freeSpace = 0;

            // if shrinks
            if (positionOfRightMenu.right > positionOfTopBar.right - 8) {
                freeSpace = positionOfTopBar.width - 16 - positionOfMoreItem.width - positionOfRightMenu.width;
            } else {
                // if extends
                freeSpace =
                    lastShowedMenuIndex === Object.keys(menuItemsRef.left).length - 1
                        ? // if only last element is hidden, calc without more menu item
                          positionOfTopBar.width - 16 - positionOfRightMenu.width
                        : positionOfTopBar.width - 16 - positionOfMoreItem.width - positionOfRightMenu.width;
            }

            // get last showed item
            menuItemsRef.left.reduce((acc, item, index) => {
                const sumWidth = acc + positionUtils.getPositionOnPage(item).width;

                if (sumWidth <= freeSpace) lastIndex = index;

                return sumWidth;
            }, 0);

            this.setState({ lastShowedMenuIndex: lastIndex });
        }
    };

    componentDidMount = () => {
        this.getMaxLeftItems();
        window.addEventListener('resize', debounce(this.getMaxLeftItems, 50));
    };

    componentWillUnmount = () => {
        window.removeEventListener('resize', debounce(this.getMaxLeftItems, 50));
    };

    render() {
        const { leftItems = [], rightItems = [], leftMenuOverTitle, colors } = this.props;
        const { mapItems, renderLeftMenu, renderRightMenu, renderShowMoreMenu } = this;
        const { lastShowedMenuIndex } = this.state;

        const showMoreItems = [{ text: leftMenuOverTitle, childDropdown: leftItems.slice(lastShowedMenuIndex + 1) }];

        return (
            <StyledTopBar
                ref={this.topBarRef}
                {...{ colors, ...this.props }}
                className={setClassName({ props: this.props, name: 'top-bar' })}>
                {leftItems.length && renderLeftMenu(mapItems(leftItems, true), colors)}
                {renderShowMoreMenu(mapItems(showMoreItems, true), colors)}
                {renderRightMenu(mapItems(rightItems, true), colors)}
            </StyledTopBar>
        );
    }
}

TopBar.displayName = 'TopBar';
TopBar.defaultProps = {
    leftMenuOverTitle: 'Other'
};

TopBar.propTypes = {
    /** Defines left items */
    leftItems: PropTypes.array,
    /** Defines left menu title */
    leftMenuOverTitle: PropTypes.string,
    /** Defines occurence */
    occurrence: PropTypes.string,
    /** Defines right items */
    rightItems: PropTypes.array,
    /** Defines component's additional styles */
    style: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string]),
    /** Defines component's additional css */
    css: PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.string])
};

export default withColors(TopBar);
