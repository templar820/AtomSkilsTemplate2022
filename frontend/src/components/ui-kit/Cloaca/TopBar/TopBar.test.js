import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import TopBar from './';

import { colors } from 'lib-root/colors';

import config from './test-config';

Enzyme.configure({ adapter: new Adapter() });

const props = {
    leftItems: config.leftItems,
    leftMenuOverTitle: 'See more...',
    occurrence: 'It',
    rightItems: config.rightItems,
    style: { background: colors.errorAccent },
    css: { fontSize: '18px' }
};

describe('<TopBar />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TopBar {...props} />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    // test must be first, otherwise the snapshot size will be too large
    test('should match snapshot', () => {
        expect(toJson(wrapper.dive().dive())).toMatchSnapshot();
    });

    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TopBar {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('should render correctly', () => {
        expect(
            wrapper
                .dive()
                .dive()
                .find('StyledLeftMenuItem').length +
                wrapper
                    .dive()
                    .dive()
                    .find('StyledShowMoreMenuItem').length +
                wrapper
                    .dive()
                    .dive()
                    .find('StyledRightMenuItem').length
        ).toBe(7); // 7 - number of menu items

        /* Checks first menu item */
        const dropdownItems = wrapper
            .dive()
            .dive()
            .find('StyledLeftMenuItem')
            .first()
            .find('withColors(DropdownItem)');

        expect(dropdownItems.length).toBe(1);
        expect(dropdownItems.props()).toHaveProperty('occurrence', props.occurrence);
    });
});
