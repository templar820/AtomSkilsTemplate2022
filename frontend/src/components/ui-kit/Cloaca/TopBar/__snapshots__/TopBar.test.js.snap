// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should match snapshot 1`] = `
<StyledTopBar
  className="CLR-top-bar"
  colors={
    Object {
      "GrayScale_0": "#ffffff",
      "GrayScale_100": "#EDEDED",
      "GrayScale_150": "#E5E5E5",
      "GrayScale_200": "#D8D8D8",
      "GrayScale_50": "#fafafa",
      "GrayScale_500": "#999999",
      "GrayScale_55": "#f6f6f6",
      "GrayScale_600": "#7C7C7C",
      "GrayScale_700": "#333333",
      "GrayScale_900": "#000000",
      "error": "#F62434",
      "errorAccent": "#EB5A40",
      "info": "#34AAF2",
      "infoAccent": "#5BD9E5",
      "primary": "#00B956",
      "primaryAccent": "#10E272",
      "secondary": "#731982",
      "secondaryAccent": "#A500BF",
      "tertiary": "#444189",
      "tertiaryAccent": "#404D46",
      "warning": "#FFA717",
      "warningAccent": "#F36F20",
    }
  }
  css={
    Object {
      "fontSize": "18px",
    }
  }
  leftItems={
    Array [
      Object {
        "childDropdown": Array [
          Object {
            "text": "Abaza and some long text",
          },
          Object {
            "text": "Abakan",
          },
        ],
        "text": "Simple item",
      },
      Object {
        "disabled": true,
        "text": "Disabled item",
      },
      Object {
        "text": "Item with text that lasts too long",
      },
      Object {
        "childDropdown": Array [
          Object {
            "text": "Abaza",
          },
          Object {
            "text": "Abakan",
          },
          Object {
            "childDropdown": Array [
              Object {
                "text": "Abaza",
              },
              Object {
                "text": "Abakan",
              },
              Object {
                "text": "Abinsk",
              },
              Object {
                "text": "Zagrabovo",
              },
            ],
            "text": "Nested dropdown",
          },
          Object {
            "text": "Abinsk",
          },
        ],
        "text": "Item with the child dropdown",
      },
      Object {
        "leftIcon": <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Autopay"
          w="25.607999999999997px"
          width={null}
        />,
        "rightIcon": <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Delete"
          w="25.607999999999997px"
          width={null}
        />,
        "text": "Item with two icons",
      },
      Object {
        "childDropdown": Array [
          Object {
            "text": "Abaza",
          },
          Object {
            "text": "Abakan",
          },
        ],
        "rightIcon": <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Archive"
          w="25.607999999999997px"
          width={null}
        />,
        "text": "Item with the child dropdown and custom right icon",
      },
    ]
  }
  leftMenuOverTitle="See more..."
  occurrence="It"
  rightItems={
    Array [
      Object {
        "childDropdown": Array [
          Object {
            "text": "First child",
          },
          Object {
            "text": "Second child",
          },
        ],
        "text": "Right item",
      },
    ]
  }
  style={
    Object {
      "background": "#EB5A40",
    }
  }
>
  <StyledLeftMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={false}
    key="0"
  >
    <withColors(DropdownItem)
      childDropdown={
        <ForwardRef(withColors(Dropdown))
          closeOnOutside={false}
          closeOnTriggerClick={true}
          disableTriggerElement={false}
          maxNumberOfVisibleOptions={6}
          renderPrimaryDropdownInPortal={true}
          scrollSnapping={true}
          tabIndex={1}
          triggerAction="click"
          triggerElement={null}
          triggerPlacement="bottom-start"
          virtualization={false}
        >
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abaza and some long text
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abakan
          </ForwardRef(withColors(DropdownItem))>
        </ForwardRef(withColors(Dropdown))>
      }
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Simple item"
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Simple item
    </withColors(DropdownItem)>
  </StyledLeftMenuItem>
  <StyledLeftMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={false}
    key="2"
  >
    <withColors(DropdownItem)
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Item with text that lasts too long"
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Item with text that lasts too long
    </withColors(DropdownItem)>
  </StyledLeftMenuItem>
  <StyledLeftMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={false}
    key="3"
  >
    <withColors(DropdownItem)
      childDropdown={
        <ForwardRef(withColors(Dropdown))
          closeOnOutside={false}
          closeOnTriggerClick={true}
          disableTriggerElement={false}
          maxNumberOfVisibleOptions={6}
          renderPrimaryDropdownInPortal={true}
          scrollSnapping={true}
          tabIndex={1}
          triggerAction="click"
          triggerElement={null}
          triggerPlacement="bottom-start"
          virtualization={false}
        >
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abaza
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abakan
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdown={
              <ForwardRef(withColors(Dropdown))
                closeOnOutside={false}
                closeOnTriggerClick={true}
                disableTriggerElement={false}
                maxNumberOfVisibleOptions={6}
                renderPrimaryDropdownInPortal={true}
                scrollSnapping={true}
                tabIndex={1}
                triggerAction="click"
                triggerElement={null}
                triggerPlacement="bottom-start"
                virtualization={false}
              >
                <ForwardRef(withColors(DropdownItem))
                  childDropdownPlacement="right-start"
                  disabled={false}
                  isOpen={true}
                  occurrence=""
                  onClick={[Function]}
                  popperProps={Object {}}
                  scrollIntoViewOptions={
                    Object {
                      "behavior": "smooth",
                      "block": "nearest",
                      "inline": "start",
                    }
                  }
                  size="16px"
                >
                  Abaza
                </ForwardRef(withColors(DropdownItem))>
                <ForwardRef(withColors(DropdownItem))
                  childDropdownPlacement="right-start"
                  disabled={false}
                  isOpen={true}
                  occurrence=""
                  onClick={[Function]}
                  popperProps={Object {}}
                  scrollIntoViewOptions={
                    Object {
                      "behavior": "smooth",
                      "block": "nearest",
                      "inline": "start",
                    }
                  }
                  size="16px"
                >
                  Abakan
                </ForwardRef(withColors(DropdownItem))>
                <ForwardRef(withColors(DropdownItem))
                  childDropdownPlacement="right-start"
                  disabled={false}
                  isOpen={true}
                  occurrence=""
                  onClick={[Function]}
                  popperProps={Object {}}
                  scrollIntoViewOptions={
                    Object {
                      "behavior": "smooth",
                      "block": "nearest",
                      "inline": "start",
                    }
                  }
                  size="16px"
                >
                  Abinsk
                </ForwardRef(withColors(DropdownItem))>
                <ForwardRef(withColors(DropdownItem))
                  childDropdownPlacement="right-start"
                  disabled={false}
                  isOpen={true}
                  occurrence=""
                  onClick={[Function]}
                  popperProps={Object {}}
                  scrollIntoViewOptions={
                    Object {
                      "behavior": "smooth",
                      "block": "nearest",
                      "inline": "start",
                    }
                  }
                  size="16px"
                >
                  Zagrabovo
                </ForwardRef(withColors(DropdownItem))>
              </ForwardRef(withColors(Dropdown))>
            }
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Nested dropdown
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abinsk
          </ForwardRef(withColors(DropdownItem))>
        </ForwardRef(withColors(Dropdown))>
      }
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Item with the child dropdown"
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Item with the child dropdown
    </withColors(DropdownItem)>
  </StyledLeftMenuItem>
  <StyledLeftMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={false}
    key="4"
  >
    <withColors(DropdownItem)
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Item with two icons"
      leftIcon={
        <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Autopay"
          w="25.607999999999997px"
          width={null}
        />
      }
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      rightIcon={
        <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Delete"
          w="25.607999999999997px"
          width={null}
        />
      }
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Item with two icons
    </withColors(DropdownItem)>
  </StyledLeftMenuItem>
  <StyledLeftMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={false}
    key="5"
  >
    <withColors(DropdownItem)
      childDropdown={
        <ForwardRef(withColors(Dropdown))
          closeOnOutside={false}
          closeOnTriggerClick={true}
          disableTriggerElement={false}
          maxNumberOfVisibleOptions={6}
          renderPrimaryDropdownInPortal={true}
          scrollSnapping={true}
          tabIndex={1}
          triggerAction="click"
          triggerElement={null}
          triggerPlacement="bottom-start"
          virtualization={false}
        >
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abaza
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Abakan
          </ForwardRef(withColors(DropdownItem))>
        </ForwardRef(withColors(Dropdown))>
      }
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Item with the child dropdown and custom right icon"
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      rightIcon={
        <ForwardRef(withColors(InlineIcons))
          color="currentColor"
          disabled={false}
          h="25.607999999999997px"
          height={null}
          icon="Archive"
          w="25.607999999999997px"
          width={null}
        />
      }
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Item with the child dropdown and custom right icon
    </withColors(DropdownItem)>
  </StyledLeftMenuItem>
  <StyledShowMoreMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    isHidden={true}
    key="0"
  >
    <withColors(DropdownItem)
      childDropdown={
        <ForwardRef(withColors(Dropdown))
          closeOnOutside={false}
          closeOnTriggerClick={true}
          disableTriggerElement={false}
          maxNumberOfVisibleOptions={6}
          renderPrimaryDropdownInPortal={true}
          scrollSnapping={true}
          tabIndex={1}
          triggerAction="click"
          triggerElement={null}
          triggerPlacement="bottom-start"
          virtualization={false}
        />
      }
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="See more..."
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      See more...
    </withColors(DropdownItem)>
  </StyledShowMoreMenuItem>
  <StyledRightMenuItem
    colors={
      Object {
        "GrayScale_0": "#ffffff",
        "GrayScale_100": "#EDEDED",
        "GrayScale_150": "#E5E5E5",
        "GrayScale_200": "#D8D8D8",
        "GrayScale_50": "#fafafa",
        "GrayScale_500": "#999999",
        "GrayScale_55": "#f6f6f6",
        "GrayScale_600": "#7C7C7C",
        "GrayScale_700": "#333333",
        "GrayScale_900": "#000000",
        "error": "#F62434",
        "errorAccent": "#EB5A40",
        "info": "#34AAF2",
        "infoAccent": "#5BD9E5",
        "primary": "#00B956",
        "primaryAccent": "#10E272",
        "secondary": "#731982",
        "secondaryAccent": "#A500BF",
        "tertiary": "#444189",
        "tertiaryAccent": "#404D46",
        "warning": "#FFA717",
        "warningAccent": "#F36F20",
      }
    }
    key="0"
  >
    <withColors(DropdownItem)
      childDropdown={
        <ForwardRef(withColors(Dropdown))
          closeOnOutside={false}
          closeOnTriggerClick={true}
          disableTriggerElement={false}
          maxNumberOfVisibleOptions={6}
          renderPrimaryDropdownInPortal={true}
          scrollSnapping={true}
          tabIndex={1}
          triggerAction="click"
          triggerElement={null}
          triggerPlacement="bottom-start"
          virtualization={false}
        >
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            First child
          </ForwardRef(withColors(DropdownItem))>
          <ForwardRef(withColors(DropdownItem))
            childDropdownPlacement="right-start"
            disabled={false}
            isOpen={true}
            occurrence=""
            onClick={[Function]}
            popperProps={Object {}}
            scrollIntoViewOptions={
              Object {
                "behavior": "smooth",
                "block": "nearest",
                "inline": "start",
              }
            }
            size="16px"
          >
            Second child
          </ForwardRef(withColors(DropdownItem))>
        </ForwardRef(withColors(Dropdown))>
      }
      childDropdownPlacement="bottom"
      childWrapperStyle={
        Object {
          "name": "1hok3eg-mapItems",
          "styles": "padding-top:5px;;label:mapItems;",
        }
      }
      disabled={false}
      isOpen={true}
      key="Right item"
      occurrence="It"
      onClick={[Function]}
      popperProps={Object {}}
      scrollIntoViewOptions={
        Object {
          "behavior": "smooth",
          "block": "nearest",
          "inline": "start",
        }
      }
      size="16px"
    >
      Right item
    </withColors(DropdownItem)>
  </StyledRightMenuItem>
</StyledTopBar>
`;
